spectator/stateless_meters.h

```diff
@@ -137,12 +137,25 @@ class DistributionSummary : public StatelessMeter<Pub> {
 template <typename Pub>
 class Gauge : public StatelessMeter<Pub> {
  public:
-  Gauge(IdPtr id, Pub* publisher)
-      : StatelessMeter<Pub>(std::move(id), publisher) {}
+  Gauge(IdPtr id, Pub* publisher, unsigned int ttl = 0)
+      : StatelessMeter<Pub>(std::move(id), publisher), ttl(ttl) {}
   void Set(double value) noexcept { this->send(value); }
 
  protected:
-  std::string_view Type() override { return "g"; }
+ // TODO: remove this when we have a better way to handle TTL
+ // this is a workaround for the fact the other Type methods
+ // return static string literals
+  std::string_view Type() override { 
+    if (ttl > 0) {
+      type_str_ = "f," + std::to_string(ttl);
+      return type_str_;
+    }
+    return "g";
```

Comment:
Should the `type_str_` default to this value, and only be changed when `ttl > 0`?